File: python/easy/#121BestTimetoBuyandSellStock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        
        profit = 0

        if len(prices) == 1:
            return 0

        minprice = prices[0]
        for i in range(0, len(prices)):
            if(prices[i] < minprice):
                minprice = prices[i]
            profit = max(profit, prices[i] - minprice)
    


        return profit

File: cpp/easy/#121BestTimetoBuyandSellStock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int maxDiff = 0;
        int currentMin;

        if(prices.size() > 1) {
            currentMin = prices[0];
        }
        
        for(int i = 1; i < prices.size(); i++) {
            if(prices[i] < currentMin) {
                currentMin = prices[i];
            }

            if(prices[i] - currentMin > maxDiff) {
                maxDiff = prices[i] - currentMin;
            }
        }


        return maxDiff;

    }
};